diff -ruN Flaky/bookkeeper-server/src/test/java/org/apache/bookkeeper/client/TestLedgerChecker.java FlakyCodeChange/bookkeeper-server/src/test/java/org/apache/bookkeeper/client/TestLedgerChecker.java
--- Flaky/bookkeeper-server/src/test/java/org/apache/bookkeeper/client/TestLedgerChecker.java	2025-05-27 21:12:41.987024538 +0600
+++ FlakyCodeChange/bookkeeper-server/src/test/java/org/apache/bookkeeper/client/TestLedgerChecker.java	2025-05-27 21:17:59.834637046 +0600
@@ -457,6 +457,7 @@
 
     private void killBookie(ArrayList<BookieSocketAddress> firstEnsemble, BookieSocketAddress ensemble)
             throws Exception {
+        try { Thread.sleep(3000);} catch (InterruptedException e) {}        
         LOG.info("Killing " + ensemble + " from ensemble=" + firstEnsemble);
         killBookie(ensemble);
     }
